torch/csrc/jit/codegen/cuda/ir_interface_nodes.h

```diff
@@ -180,20 +180,92 @@ namespace ir_utils {
 class TVDomainGuard;
 }
 
-// TensorView is our primitive Tensor Type used in code generation. It can be
-// thought of as representing physical memory, however, its dimensionality is
-// modifed as split/merge/computeAt functions are called. The history of
-// these transformations are kept and used for generating actual code referncing
-// physical memory. Generally when users are thinking of code generation in
-// reference to a Tensor, this is the class they should be interacting with.
-//
-// The reason we need both TensorView and TensorDomain is that we need to have a
-// record of both what is being computed and how it is being computed. For
-// example we may have the operation: TV3[I, J, K] = TV2[I, J, K] + TV1[I, J, K]
-// The mathematical operations here are on the tensor views TV1, TV2, and TV3.
-// This operation is a pointwise operation. To compute this pointwise operation
-// we iterate over the 3D TensorDomain [I, J, K], where K is the fastest
-// changing dimension.
+//! TensorViewOptions class is intended to be used with
+//! TensorView::makeTensor(TensorviewOptions tvo). TensorViewOptions allows
+//! users to easily set the properties of the TensorView to be constructed. This
+//! includes sizes, contiguity, number of dimensions, and type.
+struct TensorViewOptions {
+ public:
+  //! Number of dimensions of the tensor
+  int n_dims = 1;
+
+  //! Set the data type of the tensor
+  DataType dtype = DataType::Float;
+
+  //! Set if the tensor is fully contiguous. If this is set contiguity does not
+  //! need to be directly set.
+  bool is_fully_contiguous = false;
+
+  //! Set if the tensor is constructed of fully runtime sizes. If this is set,
+  //! sizes does not need to be directly set.
+  bool is_fully_symbolic = false;
+
+  //! Set the contiguity of each dimension. If specified the size of this vector
+  //! will take precedence over the ndims field.
+  std::vector<bool> contiguity;
+
+  //! Set the size of each dimension, <0 is a symbolic size, and >0 is a compile
+  //! time size. If specified the size of this vector will take precedence over
+  //! the ndims field.
+  std::vector<int64_t> sizes;
+};
+
+namespace {
+
+void validateTVO(TensorViewOptions& tvo) {
```

Comment:
naming: can we either have a more explicit name, ex. `validateTvOptions`, or just use `validate` and rely on overloading resolution?